types/base/error.d.ts

```diff
@@ -0,0 +1,21 @@
+export declare class YoniusError {
+    constructor(message: string, code?: number)
+    isClient() : Boolean
+    isServer() : Boolean
+}
+
+export declare class OperationalError extends YoniusError {
+    constructor(message?: string, code?: number)
+}
+
+export declare class NotFoundError extends OperationalError {
+    constructor(message?: string, code?: number)
+}
+
+export declare class NotImplementedError extends OperationalError {
+    constructor(message?: string, code?: number)
+}
+
+export declare class ValidationError extends OperationalError {
+    constructor(message?: string, code?: number)
+}
```

Comment:
Is this needed?

Reply:
The constructor part, or the class declaration?

Reply:
The constructor